overlord/snapstate/agentnotify/agentnotify.go

```diff
@@ -67,7 +68,8 @@ func notifyLinkSnap(snapsup *snapstate.SnapSetup) error {
 
 var sendClientFinishRefreshNotification = func(snapsup *snapstate.SnapSetup) {
 	refreshInfo := &userclient.FinishedSnapRefreshInfo{
-		InstanceName: snapsup.InstanceName(),
+		InstanceName:    snapsup.InstanceName(),
+		AppDesktopEntry: fmt.Sprintf("%s_%s", snapsup.SnapName(), snapsup.InstanceName()),
```

Comment:
This may be a bit confusing, but SnapName() is the actual snap name, and InstanceName() is snap name + instance key prefix in the context of parallel installs. That is, assuming I have both `ohmygiraffe` and `ohmygiraffe_beta` installed in the system, the strings produced here would be `ohmygiraffe_ohmygiraffe` and `ohmygiraffe_ohmygiraffe_beta` respectively. However, none of those are application names, since the applications carried by a snap are not known inside SnapSetup, and a snap can have multiple applications, each with a different icon in the desktop file.